llama_stack/apis/common/errors.py

```diff
@@ -18,38 +29,44 @@ def __init__(self, model_name: str, supported_models_list: list[str]):
         super().__init__(message)
 
 
-class ModelNotFoundError(ValueError):
+class ModelNotFoundError(ResourceNotFoundError):
     """raised when Llama Stack cannot find a referenced model"""
 
     def __init__(self, model_name: str) -> None:
-        message = f"Model '{model_name}' not found. Use client.models.list() to list available models."
-        super().__init__(message)
+        self.resource = model_name
+        self.resource_type = "Model"
+        self.client_list = "client.models.list()"
+        super().__init__(self.resource, self.resource_type, self.client_list)
 
 
-class VectorStoreNotFoundError(ValueError):
+class VectorStoreNotFoundError(ResourceNotFoundError):
     """raised when Llama Stack cannot find a referenced vector store"""
 
     def __init__(self, vector_store_name: str) -> None:
-        message = f"Vector store '{vector_store_name}' not found. Use client.vector_dbs.list() to list available vector stores."
-        super().__init__(message)
+        self.resource = vector_store_name
+        self.resource_type = "Vector Store"
+        self.client_list = "client.vector_dbs.list()"
+        super().__init__(self.resource, self.resource_type, self.client_list)
 
 
-class DatasetNotFoundError(ValueError):
+class DatasetNotFoundError(ResourceNotFoundError):
     """raised when Llama Stack cannot find a referenced dataset"""
 
     def __init__(self, dataset_name: str) -> None:
-        message = f"Dataset '{dataset_name}' not found. Use client.datasets.list() to list available datasets."
-        super().__init__(message)
+        self.resource = dataset_name
+        self.resource_type = "Dataset"
+        self.client_list = "client.datasets.list()"
+        super().__init__(self.resource, self.resource_type, self.client_list)
 
 
-class ToolGroupNotFoundError(ValueError):
+class ToolGroupNotFoundError(ResourceNotFoundError):
     """raised when Llama Stack cannot find a referenced tool group"""
 
     def __init__(self, toolgroup_name: str) -> None:
-        message = (
-            f"Tool group '{toolgroup_name}' not found. Use client.toolgroups.list() to list available tool groups."
-        )
-        super().__init__(message)
+        self.resource = toolgroup_name
+        self.resource_type = "Tool Group"
+        self.client_list = "client.toolgroups.list()"
```

Comment:
can you explain why you are assigning these instance variables? explain your reasoning and why this constitutes as reasonable software engineering.